src/utils/transactions.ts

```diff
@@ -184,3 +186,19 @@ export const getQueuedTransactionCount = (txPage?: TransactionListPage): string
 
   return queuedTxsByNonce.length.toString()
 }
+
+export const getTxOrigin = (app?: SafeAppData): string | undefined => {
+  if (!app) {
+    return
+  }
+
+  let origin: string | undefined
+
+  try {
+    origin = JSON.stringify({ name: app.name, url: app.url })
```

Comment:
Is it documented somewhere what the origin must look like? In safe-react we used `app.id` for the origin instead of a stringified object.

Reply:
The `origin` must be an object containing the `name` and `url` of the Safe App. [Here it is in `safe-react`](https://github.com/safe-global/safe-react/blob/0d09d940100ac2f0e37b371a6273b51cb490f081/src/routes/safe/components/Transactions/TxList/modals/ApproveTxModal.tsx#L164-L169). I confirmed this with the Safe Apps team yesterday.